### PR TITLE
Add error log

### PR DESCRIPTION
**Pull request by apotterri**
### Overview
The last 100 errors generated from `XMLHttpRequest` are now written to local storage. Network errors and responses with status < 500 are recorded without the response body (for the sake of brevity). The body for responses with status >= 500 are saved, as they may indicate a problem with the way the server is configure for recording.

In addition to any suggestions you may have about code or styling, I'm interested to know if:
* there is other information you think would aid in debugging
* you think 100 errors is too few or too many
* you think the different rendering of responses with status < 500 and >= 500 is useful. I did this because extensions are limited to 5MB of local storage, and the body of other responses didn't seem likely to contain interesting information.

### Walkthrough  
The error log can be retrieved by visiting the `Error Log` link from the settings page:
Screen Shot 2020-06-08 at 6 08 52 PM

Errors will have summaries that look like this:
ty-errors jpg

Network errors (i.e. those passed to an `XMLHttpRequest` `onerror` handler) will expand to this:
ty-network-error jpg
Note that neither Firefox nor Chrome provide any more details as to the cause of a network failure, so no other information is available.

HTTP errors with status >= 500 will expand to this:
ty-500-error

Other HTTP errors will be rendered without the response body.

### CODE REVIEW

**Comment by apotterri** [comment]
@ty the way the expanded details of the error are rendered doesn't seem very clear to me. I think it will be confusing, especially when there are multiple errors.

Is there some styling we can add to make them look more associated with the summary? At the very least, it seems like they should be indented a bit, and maybe have some identifiers, too:
error-log-with-keys jpg

**Comment by apotterri** [comment]
@ty not sure whether GH will send you a notification for the comment I just deleted, but you can disregard most of it....

I do think it would be helpful to have labels on the fields in the errors They don't really stand out like this, though, so maybe some  additional styling will help?
ty-errors-with-labels jpg

File: options.js [hunk 1]
```diff
@@ -68,6 +44,10 @@ export default function Options() {
     });
   };
   this.setAlternateUrl = async function(url) {
-    return writeToStorage(alternateUrlKey, url);
+    return utils.writeToStorage(alternateUrlKey, url);
+  };
+
+  this.getSaveErrors = async function() {
+    return Promise.resolve(true);
   };
```
Comment (dustinbyrne) [inline options.js:52]: Was this meant to always return true?

Comment (apotterri) [reply options.js:52]: Nope, definitely not. Good catch, thanks!

**Pull request merged**